client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -164,14 +164,14 @@ const newsletter: Flow = {
 						);
 					}
 
-					if ( providedDependencies?.goToCheckout ) {
+					if ( providedDependencies?.goToCheckout && providedDependencies?.siteSlug ) {
```

Comment:
This whole section seems like it could be abstracted to be shared across all flows?

Reply:
Yes! We did that for login redirect.